src/sql/src/plan/statement.rs

```diff
@@ -979,7 +979,15 @@ fn handle_create_type(
         let item = scx
             .catalog
             .resolve_item(&normalize::object_name(item_name)?)?;
-        ids.push(item.id());
+        let item_id = item.id();
+        if scx
+            .catalog
+            .try_get_lossy_scalar_type_by_id(&item_id)
+            .is_none()
+        {
+            bail!("{} must be data type", key);
```

Comment:
nit: error message is a little confusing if you're not sure what you're looking at; e.g., if you don't realize that `pg_enum` is a table. Maybe something like "{} is a blah, not a type" or "{} is not a type" or "{} must be data type (is a {})" or somesuch?